src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -1884,7 +1884,7 @@ public virtual bool Accepts(DType type, out KeyValuePair<string, DType> schemaDi
                         return TreeAccepts(this, TypeTree, type.TypeTree, out schemaDifference, out schemaDifferenceType, exact, useLegacyDateTimeAccepts);
                     }
 
-                    accepts = type.Kind == DKind.Unknown;
+                    accepts = (IsMultiSelectOptionSet() && TypeTree.GetPairs().First().Value.OptionSetInfo == type.OptionSetInfo) || type.Kind == DKind.Unknown;
```

Comment:
>(IsMultiSelectOptionSet() && TypeTree.GetPairs().First().Value.OptionSetInfo == type.OptionSetInfo) [](http://example.com/codeflow?start=30&length=99)

Please add a testcase in PAclient for it.

Reply:
[Added here.](https://msazure.visualstudio.com/OneAgile/_git/PowerApps-Client/pullrequest/6030875?_a=files&path=/src/Cloud/unittests/DocumentServer.Core.LargeTests/RuleToJsTest/RuleToJsTest.cs)